systemverilog-plugin/UhdmAst.cc

```diff
@@ -1028,6 +1034,45 @@ static void simplify(AST::AstNode *current_node, AST::AstNode *parent_node)
         if (current_node->str == "$display" || current_node->str == "$write")
             simplify_format_string(current_node);
         break;
+    case AST::AST_COND:
+    case AST::AST_CONDX:
+    case AST::AST_CONDZ:
+        // handle custom low high bound
+        if (current_node->attributes.count(UhdmAst::low_high_bound())) {
+            log_assert(!current_node->children.empty());
+            log_assert(current_node->children[0]->type == AST::AST_BLOCK);
+            log_assert(current_node->children[0]->children.size() == 2);
+            auto low_high_bound = current_node->children[0];
+            // this is executed when condition is met
+            // save pointer that will be added later again
+            // as conditions needs to go before this block
+            auto result = current_node->children[1];
+            current_node->children.clear();
+            while (low_high_bound->children[0]->simplify(true, false, false, 1, -1, false, false)) {
+            };
+            while (low_high_bound->children[1]->simplify(true, false, false, 1, -1, false, false)) {
+            };
+            log_assert(low_high_bound->children[0]->type == AST::AST_CONSTANT);
+            log_assert(low_high_bound->children[1]->type == AST::AST_CONSTANT);
+            const int low = low_high_bound->children[0]->integer;
+            const int high = low_high_bound->children[1]->integer;
+            int range = low_high_bound->children[0]->range_left;
+            delete low_high_bound;
+            // According to standard:
+            // If the bound to the left of the colon is greater than the
+            // bound to the right, the range is empty and contains no values.
+            for (int i = low; i >= low && i <= high; i++) {
+                // TODO(krak): what should be correct constant range if we don't have
+                // information about range?
+                if (range == -1) {
+                    range = 64;
+                }
+                current_node->children.push_back(AST::AstNode::mkconst_int(i, false, range));
+            }
```

Comment:
I believe `range_left == -1` is not an absolute indicator of invalid range. Shouldn't we check `range_valid`? Also, this ignores the range of `high`.

Also, the default should be 32:
- 32 seems to be Yosys default int constant size,
- you're storing the constant in an `int` here, so anything bigger won't work anyway,
- `mkconst_int` takes 32 bit value, so again, anything bigger won't work.